### DISCUSSION TITLE
Azure AD og antall tenant ? – Tilpasning kommunal sektor

### DISCUSSION THREAD

**Discussion by ROPEDK** [haraldfianbakken/sky-i-offentlig-sektor#19] [Show and tell]
Ser at man bør vurdere separate Azure AD tenat for forvaltning/skole/BHG. Dette er en vurdering vi har gjort for noen år tilbake. Hvor det å separere i flere tenat, antagelig ga ansatte flere utfordringer og en mer kompleks administrasjon i en multi-tenat løsning. Dette ble veldig synlig når vi også på Office365 i samme bilde. Blant annet hadde vi disse tankene med i vår vurdering: 
-       Ansatte i skole med delt stillinger i administrasjon og skole/BHG
-       Alle er ansatt i kommunen, vi så utfordringer med at ansatte kunne få flere brukerkontoer å forholde seg til. Eksempler: Er man ansatt i administrasjon/skole/BHG og løsninger for HR/lønnssytem forholder seg til et AD (On-Prem) og hva med fellesløsninger som benytter AAD for IDP.
-       Lisens administrasjon i Office365 blir pr AAD.
-       IAM system må provisjonere brukerkonto i tre AAD.

Er veldig usikker på om Tilpassingen for kommunal sektor er realistisk? For meg virker det som en veldig komplisert arkitektur. Hvor ble det av «Keep it simple» ?

**Comment by haraldfianbakken** [comment]
Strålende diskusjon og ser det bør presiseres/formuleres bedre og legge inn mer betrakninger/spørsmål:

 - Tenant for forvaltning (med delte tjenester) bør splittes ut. Dette gjelder mer for Azure tjenester og andre sentrale tjenester som kan deles og forvaltes på tvers (e.g. WAN/Nettverk/IAM mm.). Må understøtte kompleksiteten med at kommuner kan splittes eller slås sammen og forvaltning av ressuser på tvers av enheter må være enklest mulig. 
 - BHG/Skole/mm - kan godt ligge i en annen, men delt tenant (ikke nødvendigvis 1 tenant per 'enhet' som det kan fremgå som i formuleringen) - men her er betraktninger som burde tas med i forhold til innsyn/eksisterende lisensavtaler/
 - Lisensadministrasjon på tvers av AAD - så finnes det løsninger/verktøy for dette. I tillegg til at multi-tenant management stadig forbedres/kommer med ny funksjonalitet (Azure Lighthouse og M365 Lighthouse).  
 - IAM/Brukerkontoer må defintivt skisseres opp bedre - her bør man selvsagt strebe etter 1 konto og SSO for vanlige brukere. F.eks kan AD settes opp og forvaltes sentralt (i forvaltningstenant) og hver forrest/OU synkroniseres til respektive tenant. Fellesløsninger som benytter AAD må selvsagt understøtte multi-tenancy og gås opp. Her kunne vært interessant med noen eksempler så man kan grave litt/eller konkretisere dette caset ytterligere.

Keep it simple - men samtidig understøtte behovet til 80% med alle variasjoner er en interessant balanse - så det vil alltid være betraktninger/trade-offs og andre elementer som ikke vil passe for alle og variasjoner for hva som er simple eller ei. 

Dette med delte stillinger og kontoer i ulike systemer er interessant. Skiller man ikke på UPN (eller eposter) til brukere ved pålogging til e.g. administrativt system vs f.eks når de logger seg på systemet til skolen? Eller bruker de samme UPN for å logge seg på administrasjon vs skole?

Gode innspill og fint å diskusjoner rundt dette - og om vi også får konkrete applikasjoner/use-cases på - så kan det bygges ut/forbedres. 

@CrayonFT - Tagger deg også får innspill og tanker rundt temaet.
